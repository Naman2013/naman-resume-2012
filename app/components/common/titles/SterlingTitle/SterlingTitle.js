import React from 'react';
import PropTypes from 'prop-types';
import style from './SterlingTitle.style';

const SterlingTitle = ({ title, subTitle, theme }) => (
  <div className="root">
    <h4 style={theme.title} className="title">{title}</h4>
    {
      subTitle &&
        <p style={theme.subTitle} className="sub-title">{subTitle}</p>
    }
    <style jsx>{style}</style>
  </div>
);

SterlingTitle.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  theme: PropTypes.shape({
    title: PropTypes.shape({}),
    subTitle: PropTypes.shape({}),
  }),
};

SterlingTitle.defaultProps = {
  subTitle: '',
  theme: { title: {}, subTitle: {} },
};

export default SterlingTitle;
