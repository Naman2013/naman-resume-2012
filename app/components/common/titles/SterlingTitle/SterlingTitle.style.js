import css from 'styled-jsx/css';
import { primaryFont, secondaryFont } from 'styles/variables/fonts';
import { screenMedium } from 'styles/variables/breakpoints';

export default css`
  .root {
    text-align: center;
    color: #41566F;
    padding: 40px 40px 20px 40px;
  }

  .title {
    margin: 0;
    padding: 0;
    font-size: 16px;
    font-family: ${primaryFont};
    letter-spacing: 1.5px;
    text-transform: uppercase;
    margin-bottom: 10px;
    font-weight: 800;
  }

  .sub-title {
    margin: 0;
    padding: 0;
    opacity: 0;
    font-size: 16px;
    font-family: ${secondaryFont};
    transition: opacity 0.25s ease-in;
    text-transform: capitalize;
  }

  @media ${screenMedium} {
    .root {
      padding-top: 60px;
      padding-bottom: 40px;
    }

    .sub-title {
      opacity: 1;
    }
  }
`;
