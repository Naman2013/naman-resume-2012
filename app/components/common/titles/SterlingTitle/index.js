import SterlingTitle from './SterlingTitle';

export default SterlingTitle;
