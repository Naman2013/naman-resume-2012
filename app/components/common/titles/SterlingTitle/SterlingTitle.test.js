import React from 'react';
import { shallow } from 'enzyme';
import SterlingTitle from './SterlingTitle';

describe('SterlingTitle', () => {

  const setup = () => {
    const props = { title: 'Foo', subTitle: 'Bar' };
    const shallowWrapper = shallow(<SterlingTitle {...props} />);
    return {
      shallowWrapper,
      subTitle: shallowWrapper.find('.sub-title'),
    };
  };

  it('should render correctly', () => {
    expect(setup().shallowWrapper).toMatchSnapshot();
  });

  describe('when providing a subTitle', () => {
    it('should render a subTitle when it is provided', () => {
      expect(setup().subTitle.length).toEqual(1);
    });

    it('should not render a sub title when NOT provided', () => {
      expect(setup().shallowWrapper.setProps({ subTitle: '' }).find('.sub-title').length).toEqual(0);
    });
  });
});
