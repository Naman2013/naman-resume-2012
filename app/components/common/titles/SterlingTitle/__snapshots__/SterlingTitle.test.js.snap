// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SterlingTitle should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SterlingTitle
    subTitle="Bar"
    theme={
      Object {
        "subTitle": Object {},
        "title": Object {},
      }
    }
    title="Foo"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4
          className="jsx-680802847 title"
          style={Object {}}
        >
          Foo
        </h4>,
        <p
          className="jsx-680802847 sub-title"
          style={Object {}}
        >
          Bar
        </p>,
        <JSXStyle
          css={
            Array [
              ".root.jsx-680802847{text-align:center;color:#41566F;padding:40px 40px 20px 40px;}",
              ".title.jsx-680802847{margin:0;padding:0;font-size:16px;font-family:brandon-grotesque, sans-serif;-webkit-letter-spacing:1.5px;-moz-letter-spacing:1.5px;-ms-letter-spacing:1.5px;letter-spacing:1.5px;text-transform:uppercase;margin-bottom:10px;font-weight:800;}",
              ".sub-title.jsx-680802847{margin:0;padding:0;opacity:0;font-size:16px;font-family:adobe-garamond-pro, serif;-webkit-transition:opacity 0.25s ease-in;transition:opacity 0.25s ease-in;}",
              "@media screen and (min-width: 768px){.root.jsx-680802847{padding-top:60px;padding-bottom:40px;}.sub-title.jsx-680802847{opacity:1;}}",
            ]
          }
          styleId="680802847"
        />,
      ],
      "className": "jsx-680802847 root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Foo",
          "className": "jsx-680802847 title",
          "style": Object {},
        },
        "ref": null,
        "rendered": "Foo",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Bar",
          "className": "jsx-680802847 sub-title",
          "style": Object {},
        },
        "ref": null,
        "rendered": "Bar",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            ".root.jsx-680802847{text-align:center;color:#41566F;padding:40px 40px 20px 40px;}",
            ".title.jsx-680802847{margin:0;padding:0;font-size:16px;font-family:brandon-grotesque, sans-serif;-webkit-letter-spacing:1.5px;-moz-letter-spacing:1.5px;-ms-letter-spacing:1.5px;letter-spacing:1.5px;text-transform:uppercase;margin-bottom:10px;font-weight:800;}",
            ".sub-title.jsx-680802847{margin:0;padding:0;opacity:0;font-size:16px;font-family:adobe-garamond-pro, serif;-webkit-transition:opacity 0.25s ease-in;transition:opacity 0.25s ease-in;}",
            "@media screen and (min-width: 768px){.root.jsx-680802847{padding-top:60px;padding-bottom:40px;}.sub-title.jsx-680802847{opacity:1;}}",
          ],
          "styleId": "680802847",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            className="jsx-680802847 title"
            style={Object {}}
          >
            Foo
          </h4>,
          <p
            className="jsx-680802847 sub-title"
            style={Object {}}
          >
            Bar
          </p>,
          <JSXStyle
            css={
              Array [
                ".root.jsx-680802847{text-align:center;color:#41566F;padding:40px 40px 20px 40px;}",
                ".title.jsx-680802847{margin:0;padding:0;font-size:16px;font-family:brandon-grotesque, sans-serif;-webkit-letter-spacing:1.5px;-moz-letter-spacing:1.5px;-ms-letter-spacing:1.5px;letter-spacing:1.5px;text-transform:uppercase;margin-bottom:10px;font-weight:800;}",
                ".sub-title.jsx-680802847{margin:0;padding:0;opacity:0;font-size:16px;font-family:adobe-garamond-pro, serif;-webkit-transition:opacity 0.25s ease-in;transition:opacity 0.25s ease-in;}",
                "@media screen and (min-width: 768px){.root.jsx-680802847{padding-top:60px;padding-bottom:40px;}.sub-title.jsx-680802847{opacity:1;}}",
              ]
            }
            styleId="680802847"
          />,
        ],
        "className": "jsx-680802847 root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Foo",
            "className": "jsx-680802847 title",
            "style": Object {},
          },
          "ref": null,
          "rendered": "Foo",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bar",
            "className": "jsx-680802847 sub-title",
            "style": Object {},
          },
          "ref": null,
          "rendered": "Bar",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              ".root.jsx-680802847{text-align:center;color:#41566F;padding:40px 40px 20px 40px;}",
              ".title.jsx-680802847{margin:0;padding:0;font-size:16px;font-family:brandon-grotesque, sans-serif;-webkit-letter-spacing:1.5px;-moz-letter-spacing:1.5px;-ms-letter-spacing:1.5px;letter-spacing:1.5px;text-transform:uppercase;margin-bottom:10px;font-weight:800;}",
              ".sub-title.jsx-680802847{margin:0;padding:0;opacity:0;font-size:16px;font-family:adobe-garamond-pro, serif;-webkit-transition:opacity 0.25s ease-in;transition:opacity 0.25s ease-in;}",
              "@media screen and (min-width: 768px){.root.jsx-680802847{padding-top:60px;padding-bottom:40px;}.sub-title.jsx-680802847{opacity:1;}}",
            ],
            "styleId": "680802847",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
